src/Atmos/Model/moisture.jl

```diff
@@ -0,0 +1,95 @@
+#### Moisture component in atmosphere model
+abstract type MoistureModel end
+
+vars_state(::MoistureModel, T) = Tuple{}
+vars_transform(::MoistureModel, T) = Tuple{}
+vars_diffusive(::MoistureModel, T) = Tuple{}
+vars_aux(::MoistureModel, T) = Tuple{}
+
+function diffusive!(::MoistureModel, diffusive, ∇transform, state, aux, t, ν)
+end
+function flux_diffusive!(::MoistureModel, flux::Grad, state::Vars, diffusive::Vars, aux::Vars, t::Real)
+end
+function update_aux!(::MoistureModel, state::Vars, diffusive::Vars, aux::Vars, t::Real)
+end
+function gradtransform!(::MoistureModel, transform::Vars, state::Vars, aux::Vars, t::Real)
+end
+
+
+
+struct EquilMoist <: MoistureModel
+end
+vars_state(::EquilMoist,T) = NamedTuple{(:ρq_tot,),Tuple{T}}
+vars_transform(::EquilMoist,T) = NamedTuple{(:q_tot),Tuple{T}}
+vars_diffusive(::EquilMoist,T) = NamedTuple{(:ρd_q_tot, :ρJ_ρD), Tuple{SVector{3,T},SVector{3,T}}}
+vars_aux(::EquilMoist,T) = NamedTuple{(:e_int, :temperature),Tuple{T,T}}
+
+
+function pressure(::EquilMoist, state::Vars, aux::Vars, t::Real)
+  ρ = state.ρ
+  q_tot = stat.ρq_tot / ρ
+  phase = PhaseEquil(aux.moisture.e_int, q_tot, ρ, aux.moisture.temperature)
+  air_pressure(phase)
+end
+function soundspeed(::EquilMoist, state::Vars, aux::Vars, t::Real)
+  ρ = state.ρ
+  q_tot = stat.ρq_tot / ρ
+  phase = PhaseEquil(aux.moisture.e_int, q_tot, ρ, aux.moisture.temperature)
+  soundspeed_air(phase)
+end
+
+function gradtransform!(::EquilMoist, transform::Vars, state::Vars, aux::Vars, t::Real)
+  ρinv = 1 / state.ρ
+  transform.moisture.q_tot = ρinv * state.moisture.ρq_tot
+  transform.moisture.temperature = aux.moisture.temperature
+end
+
+
+function diffusive!(::EquilMoist, diffusive::Vars, ∇transform::Grad, state::Vars, aux::Vars, t::Real, ν::Union{Real,AbstractMatrix})
+  # turbulent Prandtl number
+  diag_ν = ν isa Real ? ν : diag(ν) # either a scalar or vector
+  D_q_tot = diag_ν / Prandtl_t # either a scalar or vector
+  
+  # diffusive flux of q_tot
+  diffusive.moisture.ρd_q_tot = state.ρ * (-D_q_tot) .* ∇transform.moisture.q_tot # a vector
+
+  # J is the conductive or SGS turbulent flux of sensible heat per unit mass
+  ρJ = state.ρ * (cp_d / Prandtl_t) * ν * ∇transform.moisture.temperature
```

Comment:
Better to explicitly use same diffusivity as for moisture, in place of `\nu / Prandtl_t`.

Reply:
You'd want the moist thermodynamic heat capacity here, and `cv` in place of `cp`, so `cv_m` from `MoistThermodynamics`